### ISSUE TITLE
simplex mesh / dmplex

### ISSUE THREAD

**Issue by lmoresi** [underworldcode/underworld3-archive#68]
After the mesh tidying commit, I am getting errors from this:

```
        self.isSimplex = self.dm.isSimplex()
        self.cdim = self.dm.getDimension()

        # Use grid hashing for point location
        options = PETSc.Options()
        options["dm_plex_hash_location"] = 0
        self.dm.setFromOptions()

```

because `dm.isSimplex()` does not seem to exist (or be wrapped) in 3.16. Can we make a compatibility wrapper for this @rbeucher / @julesghub ?

**Comment by lmoresi** [comment]
Also, the cdim argument was supposed to allow the embedding of manifolds - is that now automatic in dmplex ?

**Comment by knepley** [comment]
I think you might want
```
self.cdim  = self.dm.getCoordinateDim()
```
Let me see about isSimplex.

**Comment by lmoresi** [comment]
Thanks @knepley - that's what romain has used (hooray !). I bet we have a bunch more questions about this particular feature pretty soon. 

isSimplex obviously works for @rbeucher in 3.17 but seems to be missing from 3.16 ... though we always access via petsc4py so we sometimes are just missing the wrapper.

**Comment by knepley** [comment]
Oh yes, I only put it in for this release.

**Comment by lmoresi** [comment]
Well, it's very handy, better than having to define these things and keep them in sync.

**Comment by lmoresi** [comment]
Also, @knepley, while we have you here, the documentation for periodic meshes gmsh->dmplex is not extensive and that could mean "it just works" or it might mean the opposite. Is there anything we need to know ?

**Comment by knepley** [comment]
Here is how I do it. I read in the periodic Gmsh, and form an actual periodic topology, but have a DG coordinate field hidden, so there can be jumps.

Jed is proposing that we have an alternative scheme that has a non-periodic topology, but identification of dofs during the local-to-global map. I don't like this because it breaks the topological queries, but he likes it because it enables strange stuff like half-rotations.

Tl;DR: I think it should just work for you

**Comment by lmoresi** [comment]
OK, watch this space, we'll be right back.

**Comment by lmoresi** [comment]
I think this is ok to close. We have most of what we need. The periodic stuff is a bit hard to follow but I think I get it now.
